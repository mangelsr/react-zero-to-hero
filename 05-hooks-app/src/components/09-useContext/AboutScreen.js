import React, { useContext } from 'react'
import { UserContext } from './UserContext'

export const AboutScreen = () => {

    const { user, setUser } = useContext(UserContext);

    const handleClick = () => {
        setUser({});
    };

    return (
        <div>
            <h1>About Screen</h1>
            <hr/>

            <pre>
                { JSON.stringify(user, null, 3) }
            </pre>

            <button
                onClick={handleClick}
                className="btn btn-outline-danger">
                Logout
            </button>
        </div>
    )
}
