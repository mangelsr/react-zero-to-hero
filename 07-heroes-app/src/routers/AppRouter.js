import React, { useContext } from 'react';
import {
    BrowserRouter as Router,
    Switch
} from "react-router-dom";

import { AuthContext } from '../auth/AuthContext';
import { LoginScreen } from '../components/login/LoginScreen';
import { DashboardRoutes } from './DashboardRoutes';
import { PrivateRoute } from './PrivateRoute';
import { PublicRoute } from './PublicRoute';


export const AppRouter = () => {

    const { user:{logged} } = useContext(AuthContext);

    return (
        <div>
            <Router>
            <div>
                <Switch>
                    <PublicRoute isAuth={logged} exact path="/login" component={LoginScreen} />
                    <PrivateRoute isAuth={logged} path="/" component={DashboardRoutes} />
                </Switch>
            </div>
            </Router>
        </div>
    )
}
