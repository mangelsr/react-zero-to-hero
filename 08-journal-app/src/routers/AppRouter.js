import React, { useEffect, useState } from 'react';
import { BrowserRouter as Router, Switch, Redirect } from 'react-router-dom';
import { useDispatch } from 'react-redux';

import { AuthRouter } from './AuthRouter';
import { JournalScreen } from '../components/journal/JournalScreen';
import { firebase } from "../firebase/firebaseConfig";
import { login } from '../actions/auth';
import { PublicRoute } from './PublicRoute';
import { PrivateRoute } from './PrivateRoute';
import { startLoadingNotes } from '../actions/notes';


export const AppRouter = () => {

    const dispatch = useDispatch();
    const [checking, setChecking] = useState(true);
    const [isLoggedIn, setIsLoggedIn] = useState(false);

    useEffect(() => {
        firebase.auth().onAuthStateChanged(async (user) => {
            if (user?.uid) {
                dispatch(login(user.uid, user.displayName));
                setIsLoggedIn(true);
                dispatch(startLoadingNotes(user.uid)) ;
            } else {
                setIsLoggedIn(false);
            }
            setChecking(false);
        });
    }, [dispatch, setChecking, setIsLoggedIn]);

    if (checking) {
        return <h1>Wait...</h1>;
    } else {
        return (
            <Router>
                <Switch>
                    <PublicRoute isAuth={isLoggedIn} path="/auth" component={AuthRouter} />
                    <PrivateRoute exact isAuth={isLoggedIn} path="/" component={JournalScreen} />
                    <Redirect to="/" />
                </Switch>
            </Router>
        );
    }
}
