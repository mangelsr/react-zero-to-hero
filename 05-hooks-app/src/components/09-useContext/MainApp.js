import React, { useEffect, useState } from 'react';
import { AppRouter } from './AppRouter';

import { UserContext } from './UserContext';
import './style.css';


let initialUser = JSON.parse(localStorage.getItem('user'));
if (initialUser === null) {
    initialUser = {};
}

export const MainApp = () => {

    const [user, setUser] = useState(initialUser);

    useEffect(() => {
        localStorage.setItem('user', JSON.stringify(user));
    }, [user]);

    return (
        <UserContext.Provider value={{ user, setUser }}>
            <AppRouter/>
        </UserContext.Provider>   
    )
}
