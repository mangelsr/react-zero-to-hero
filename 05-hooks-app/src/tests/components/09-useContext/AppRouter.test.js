import '@testing-library/jest-dom';
import { mount } from 'enzyme';
import React from 'react';
import { AppRouter } from '../../../components/09-useContext/AppRouter';
import { UserContext } from '../../../components/09-useContext/UserContext';


describe('AppRouter component tests', () => {
   
    const user = {
        id: 1,
        name: 'Miguel'
    };
    
    const wrapper = mount(
        <UserContext.Provider value={{
            user
        }}>
            <AppRouter />
        </UserContext.Provider>
    
    );

    test('should render the component', () => {
        expect(wrapper).toMatchSnapshot();
    });

});