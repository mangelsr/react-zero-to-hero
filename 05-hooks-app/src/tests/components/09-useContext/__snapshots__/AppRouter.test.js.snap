// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppRouter component tests should render the component 1`] = `
<div>
  <nav
    className="navbar navbar-expand-md navbar-dark bg-dark"
  >
    <a
      className="navbar-brand"
      href="/"
      onClick={[Function]}
    >
      useContext
    </a>
    <button
      aria-controls="navbarNav"
      aria-expanded="false"
      aria-label="Toggle navigation"
      className="navbar-toggler"
      data-target="#navbarNav"
      data-toggle="collapse"
      type="button"
    >
      <span
        className="navbar-toggler-icon"
      />
    </button>
    <div
      className="collapse navbar-collapse"
      id="navbarNav"
    >
      <ul
        className="navbar-nav"
      >
        <a
          aria-current="page"
          className="nav-item nav-link active"
          href="/"
          onClick={[Function]}
          style={Object {}}
        >
          Home
        </a>
        <a
          aria-current={null}
          className="nav-item nav-link"
          href="/about"
          onClick={[Function]}
        >
          About
        </a>
        <a
          aria-current={null}
          className="nav-item nav-link"
          href="/login"
          onClick={[Function]}
        >
          Login
        </a>
      </ul>
    </div>
  </nav>
  <div
    className="container"
  >
    <div>
      <h1>
        Home Screen
      </h1>
      <hr />
      <pre>
        {
   "id": 1,
   "name": "Miguel"
}
      </pre>
    </div>
  </div>
</div>
`;
